src/gateway/util/Send.ts

```diff
@@ -7,8 +7,23 @@ try {
 	);
 }
 import { Payload, WebSocket } from "@fosscord/gateway";
+import fs from "fs";
+import path from "path";
 
 export function Send(socket: WebSocket, data: Payload) {
+	if (process.env.WS_VERBOSE) console.log(`[Websocket] Outgoing message: ${JSON.stringify(data)}`);
+	if (process.env.WS_DUMP) {
+		if(socket.session_id) {
+			fs.mkdirSync(path.join("dump", socket.session_id), { recursive: true });
+			fs.writeFileSync(path.join("dump", socket.session_id, `${Date.now()}.out.json`), JSON.stringify(data, null, 2));
+		}
+		else {
+			fs.mkdirSync(path.join("dump", "unknown"), { recursive: true });
+			fs.writeFileSync(path.join("dump", "unknown", `${Date.now()}.out.json`), JSON.stringify(data, null, 2));
```

Comment:
Same deal with sync fs calls

Reply:
this isnt an async function, Maddy...

Reply:
You don't need to await it. Either define an async function which is immediately called or do it in the returned promise.